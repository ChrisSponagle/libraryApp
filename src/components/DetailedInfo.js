import React, { Component } from 'react';
import { View, Text, StyleSheet, Dimensions, TouchableOpacity, Image, ScrollView } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import Markdown from 'react-native-markdown-renderer';
import Header from './Header';
// import item from './mockItem.json' **For jest testing**

function DetailedInfo(props) {
    const { item } = props.navigation.state.params
    return (
        <View style={styles.container}>
            <LinearGradient
                colors={['rgba(231, 122, 52, 0.89)', 'rgba(203, 204, 203, 0.7565)']}
                start={[0, 0.5]}
                style={styles.backgroundGradient}
            />
            <TouchableOpacity style={styles.backArrowContainer} onPress={() => props.navigation.goBack() }>
                <Image style={styles.backArrowImg} source={require('../assets/images/backArrow.png')}/>
            </TouchableOpacity>
            <Header type={'detailed'}/>
            <View style={styles.detailedInfoContainer}>
                <ScrollView contentContainerStyle={styles.scrollContainer}>
                    <Text style={styles.title}>{item.title}</Text>
                    <Text style={styles.author}>{item.author.name}</Text>
                    <Markdown>{item.body}</Markdown>
                </ScrollView>
            </View>
        </View>
    )
}

const screenWidth = Math.round(Dimensions.get('window').width);
const screenHeight = Math.round(Dimensions.get('window').height);

const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
    },
    backgroundGradient: {
      position: 'absolute',
      left: 0,
      right: 0,
      top: 0,
      height: screenHeight,
    },
    backArrowContainer: {
      position: 'absolute',
      left: 10,
      top: 40,
      height: 50,
      alignItems: 'center',
      justifyContent: 'center',
      zIndex: 3
      
    },
    backArrowImg: {
      width: 30,
      height: 30
    },
    detailedInfoContainer: {
        width: screenWidth - 50,
        height: screenHeight/1.6,
        padding: 15,
        marginBottom: 100,
        backgroundColor: '#FFF',
        alignItems: 'center',
        borderRadius: 4,
        marginTop: 30,
        elevation: 5, //For Android 
        shadowColor: '#000', //For iOS
        shadowOffset: { width: 0, height: 1 },
        shadowOpacity: 0.8,
        shadowRadius: 1
    },
    scrollContainer: {
        alignItems: 'center'
    },
    title: {
        fontSize: 30, 
        color: '#E77A34'
    },
    author: {
        fontSize: 16,
        color: '#000000',
        fontWeight: 'bold'
    },
    body: {
        fontSize: 14,
        color: '#000000'
    }
  });

  export default DetailedInfo;