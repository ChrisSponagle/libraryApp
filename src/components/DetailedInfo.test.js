import React from 'react';
import renderer from 'react-test-renderer';
import { enableFetchMocks } from 'jest-fetch-mock';
import * as navigation from "react-navigation";

import DetailedInfo from './DetailedInfo';

enableFetchMocks()
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('<DetailedInfo />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<DetailedInfo />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});