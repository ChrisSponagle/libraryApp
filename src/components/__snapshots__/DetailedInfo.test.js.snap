// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailedInfo /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          3823598132,
          3251358923,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": Array [
          0,
          0.5,
        ],
      }
    }
    style={
      Object {
        "height": 1334,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "left": 10,
        "opacity": 1,
        "position": "absolute",
        "top": 40,
        "zIndex": 3,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 160,
        "padding": 15,
        "width": 750,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#FFF",
          "fontSize": 30,
        }
      }
    >
      silverorange Library
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 4,
        "elevation": 5,
        "height": 833.75,
        "marginBottom": 100,
        "marginTop": 30,
        "padding": 15,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 1,
        "width": 700,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#E77A34",
              "fontSize": 30,
            }
          }
        >
          Salvage
        </Text>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Bill Johnson
        </Text>
        <View
          style={Object {}}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {},
                  Object {
                    "fontSize": 32,
                  },
                ]
              }
            >
              <Text>
                In et illo
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {},
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
            >
              <Text>
                Movere ora terque baculum colebatur
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-start",
                "marginBottom": 10,
                "marginTop": 10,
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text>
                Lorem markdownum solito, vocat nata corripuere semper mitia tosta. Et ductae
              </Text>
              <Text>
                

              </Text>
              <Text>
                ingreditur saevit vero penates laevum, et mihi longum, incalfacit est
              </Text>
              <Text>
                

              </Text>
              <Text>
                commissaque, tuae locum. Redeuntis sulcat non magna
              </Text>
              <Text>
                

              </Text>
              <Text
                onPress={[Function]}
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                <Text>
                  Sticteque
                </Text>
              </Text>
              <Text>
                , tua, ut Oceani. Mediusque adest
              </Text>
              <Text>
                

              </Text>
              <Text>
                sacra, 
              </Text>
              <Text
                onPress={[Function]}
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                <Text>
                  aut
                </Text>
              </Text>
              <Text>
                 iam Somnus nunc. Una facinus tibi
              </Text>
              <Text>
                

              </Text>
              <Text>
                praelata via vestem vidit minorem tacto cui?
              </Text>
            </Text>
          </View>
          <Text
            style={
              Object {
                "backgroundColor": "#f5f5f5",
                "borderColor": "#CCCCCC",
                "borderRadius": 4,
                "borderWidth": 1,
                "padding": 10,
              }
            }
          >
            if (southbridge(rate_file)) {
    streaming(on_mirrored - copyright, 118971);
    internic = right_hibernate;
    nativeSimm.pinterestCloudPpga(primary);
}
eide_number(memory_mbps_rosetta(serverDvd - oem, -2, 1));
hard_subdirectory = 97 - 18;
if (icqMacintoshAlert == vdslHypermedia) {
    nvramCableColumn = data_friendly(subnetScannerSocial + cms, frameMetal,
            scraping_skyscraper);
    firmwareControlMac(e_im, operation_telnet_metafile(3, javaDynamicKernel,
            publishing_standalone_undo));
    shortcut_newline_hard += netmask_signature;
}

          </Text>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-start",
                "marginBottom": 10,
                "marginTop": 10,
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text
                onPress={[Function]}
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                <Text>
                  Iubemur agris equo
                </Text>
              </Text>
              <Text>
                 omne 
              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                <Text>
                  curruque
                </Text>
              </Text>
              <Text>
                 proxima, meae nam,
              </Text>
              <Text>
                

              </Text>
              <Text
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                <Text>
                  adpropera
                </Text>
              </Text>
              <Text>
                 videt, litora novum, Sparte. Repetit terra, ducere prosiliunt me
              </Text>
              <Text>
                

              </Text>
              <Text>
                regis: Hippolytus iam poteram. Adsumere et tuque, mora tenet per et de hominis.
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {},
                  Object {},
                  Object {
                    "fontSize": 24,
                  },
                ]
              }
            >
              <Text>
                Supplex ardescit aequore rebus Hersilien exspectata ille
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-start",
                "marginBottom": 10,
                "marginTop": 10,
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text>
                Esse patrioque istis studiosius frui: laeta cum, facit caelum. Spiramenta furca
              </Text>
              <Text>
                

              </Text>
              <Text>
                eris electus quod 
              </Text>
              <Text
                style={
                  Object {
                    "fontStyle": "italic",
                  }
                }
              >
                <Text>
                  semper sanguine infundere
                </Text>
              </Text>
              <Text>
                 illi: forte iram nec, in pietas
              </Text>
              <Text>
                

              </Text>
              <Text>
                quietis orbem, erat. Non virtute ardere cacumen perdere iramque iacuit super
              </Text>
              <Text>
                

              </Text>
              <Text>
                senior placidissime caelo? Per tenebat supervolat filia damnandus, Epiros nec
              </Text>
              <Text>
                

              </Text>
              <Text>
                pro perstet.
              </Text>
            </Text>
          </View>
          <View
            style={
              Array [
                Object {},
                Object {},
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text
                style={
                  Object {
                    "lineHeight": 36,
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              >
                ·
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexWrap": "wrap",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                      "justifyContent": "flex-start",
                      "marginBottom": 10,
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={Object {}}
                  >
                    <Text>
                      Nobis et temerat seductus
                    </Text>
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text
                style={
                  Object {
                    "lineHeight": 36,
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              >
                ·
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexWrap": "wrap",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                      "justifyContent": "flex-start",
                      "marginBottom": 10,
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={Object {}}
                  >
                    <Text>
                      Tanto murra sanguinis humus victore imo amoris
                    </Text>
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                }
              }
            >
              <Text
                style={
                  Object {
                    "lineHeight": 36,
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              >
                ·
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "flexWrap": "wrap",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                      "justifyContent": "flex-start",
                      "marginBottom": 10,
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={Object {}}
                  >
                    <Text>
                      Et et feri rursus
                    </Text>
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-start",
                "marginBottom": 10,
                "marginTop": 10,
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text>
                Et tamen color; sacerdos est inpubibus
              </Text>
              <Text>
                

              </Text>
              <Text
                onPress={[Function]}
                style={
                  Object {
                    "textDecorationLine": "underline",
                  }
                }
              >
                <Text>
                  angues
                </Text>
              </Text>
              <Text>
                 est nec lenti palluit, tantum.
              </Text>
              <Text>
                

              </Text>
              <Text>
                Blandas Oliaros tuque pater! Hunc prima natarum, ira nocebant hasta filia, suo,
              </Text>
              <Text>
                

              </Text>
              <Text>
                nos sua signa, praepetis dixit tamen.
              </Text>
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "flex-start",
                "marginBottom": 10,
                "marginTop": 10,
              }
            }
          >
            <Text
              style={Object {}}
            >
              <Text>
                Caput videam est dixit nisi, thracius praesaga una aliter, dimittit! Et
              </Text>
              <Text>
                

              </Text>
              <Text>
                fortisque talia et pingues nulloque vivacia dumque haberet fuit clamavit minus.
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
